### PR TITLE
Refactor to add new Elasticsearch API Client

### PR DESCRIPTION
**Pull request by jdrossl**
https://github.com/craftercms/craftercms/issues/5210

Depends on https://github.com/craftercms/search/pull/267

### CODE REVIEW

File: src/main/java/org/craftercms/deployer/utils/elasticsearch/AbstractElasticsearchFactory.java [hunk 1]
```diff
@@ -74,20 +74,20 @@ protected T createInstance() {
...
      * @return the service instance
      */
-    protected abstract T doCreateMultiInstance(RestHighLevelClient readClient, RestHighLevelClient[] writeClients);
+    protected abstract T doCreateMultiInstance(ElasticsearchClient readClient, ElasticsearchClient[] writeClients);
 
     @Override
     protected void destroyInstance(final T instance) throws Exception {
-        logger.debug("Closing all clients for '{}'", name);
-        instance.close();
+//        logger.debug("Closing all clients for '{}'", name);
```
Comment (sumerjabri) [inline src/main/java/org/craftercms/deployer/utils/elasticsearch/AbstractElasticsearchFactory.java]: Why keep these around?

File: src/main/java/org/craftercms/deployer/utils/elasticsearch/legacy/AbstractElasticsearchFactory.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (C) 2007-2020 Crafter Software Corporation. All Rights Reserved.
```
Comment (sumerjabri) [inline src/main/java/org/craftercms/deployer/utils/elasticsearch/legacy/AbstractElasticsearchFactory.java]: Please update to 2022

File: src/main/java/org/craftercms/deployer/utils/elasticsearch/legacy/ElasticsearchClusterConfig.java [hunk 3]
```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (C) 2007-2020 Crafter Software Corporation. All Rights Reserved.
```
Comment (sumerjabri) [inline src/main/java/org/craftercms/deployer/utils/elasticsearch/legacy/ElasticsearchClusterConfig.java]: 2022

File: src/main/java/org/craftercms/deployer/utils/elasticsearch/legacy/ElasticsearchConfig.java [hunk 4]
```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (C) 2007-2020 Crafter Software Corporation. All Rights Reserved.
```
Comment (sumerjabri) [inline src/main/java/org/craftercms/deployer/utils/elasticsearch/legacy/ElasticsearchConfig.java]: 2022

**Review by sumerjabri — CHANGES_REQUESTED** [review]

**Review by avasquez614 — APPROVED** [review]

**Review by sumerjabri — APPROVED** [review]

**Pull request merged**